youtube_dl/extractor/ciscolive.py

```diff
@@ -21,19 +22,35 @@ class CiscoLiveBaseIE(InfoExtractor):
     # These appear to be constant across all Cisco Live presentations
     # and are not tied to any user session or event
     RAINFOCUS_API_URL = 'https://events.rainfocus.com/api/%s'
-    RAINFOCUS_API_PROFILE_ID = 'Na3vqYdAlJFSxhYTYQGuMbpafMqftalz'
-    RAINFOCUS_WIDGET_ID = 'n6l4Lo05R8fiy3RpUBm447dZN8uNWoye'
+    RAINFOCUS_API_PROFILE_ID = ''  # if blank will be fetched at runtime from site javascript
+    RAINFOCUS_WIDGET_ID = ''  # if blank will be fetched at runtime from site javascript
+    RAINFOCUS_TOKENS_URL = 'https://cdn-events.rainfocus.com/pages/cisco/clondemand/catalog.js'
     BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/5647924234001/SyK2FdqjM_default/index.html?videoId=%s'
 
     HEADERS = {
-        'Origin': 'https://ciscolive.cisco.com',
+        'Origin': 'https://ciscolive.com',
         'rfApiProfileId': RAINFOCUS_API_PROFILE_ID,
         'rfWidgetId': RAINFOCUS_WIDGET_ID,
     }
 
     def _call_api(self, ep, rf_id, query, referrer, note=None):
         headers = self.HEADERS.copy()
         headers['Referer'] = referrer
+        if not self.RAINFOCUS_API_PROFILE_ID and not self.RAINFOCUS_WIDGET_ID:
+            rf_token_js = self._download_webpage(self.RAINFOCUS_TOKENS_URL, 'catalog.js', headers=headers)
+            if "apiToken" not in rf_token_js:
+                raise ExtractorError(
+                    'Unable to fetch api profile (apiToken) value.', expected=True)
+            if "widgetId" not in rf_token_js:
+                raise ExtractorError(
+                    'Unable to fetch widgetId value', expected=True)
+            api_token = self._html_search_regex(
+                r'(?:apiToken): \'(\w+)\',', rf_token_js, 'apiToken')
+            widget_id = self._html_search_regex(
+                r'(?:widgetId): \'(\w+)\',', rf_token_js, 'widgetId')
+            if api_token and widget_id:
+                headers['rfApiProfileId'] = api_token
+                headers['rfWidgetId'] = widget_id
```

Comment:
The (non-empty) matches succeeded (or an exception would have been raised) so the values are both truthy:
```suggestion
            headers['rfApiProfileId'] = api_token
            headers['rfWidgetId'] = widget_id
```

Reply:
updated in 9409e00 as suggested